packages/infrastructure/kube_external_dns/main.tf

```diff
@@ -1,115 +1,50 @@
 terraform {
-  required_providers {
-    kubernetes = {
-      source  = "hashicorp/kubernetes"
-      version = "2.27.0"
-    }
-    helm = {
-      source  = "hashicorp/helm"
-      version = "2.12.1"
-    }
-    aws = {
-      source  = "hashicorp/aws"
-      version = "5.39.1"
-    }
-    random = {
-      source  = "hashicorp/random"
-      version = "3.6.0"
-    }
-    kubectl = {
-      source  = "alekc/kubectl"
-      version = "2.0.4"
-    }
-  }
+  required_providers {}
 }
 
 locals {
-
   name      = "external-dns"
   namespace = module.namespace.namespace
-
-  all_roles = toset([for domain, config in var.route53_zones : config.record_manager_role_arn])
-  config = { for role in local.all_roles : role => {
-    labels           = { role : sha1(role) }
-    zone_ids         = [for domain, config in var.route53_zones : config.zone_id if config.record_manager_role_arn == role]
-    included_domains = [for domain, config in var.route53_zones : domain if config.record_manager_role_arn == role]
-    excluded_domains = [for domain, config in var.route53_zones : domain if config.record_manager_role_arn != role && alltrue([for includedDomain, config in var.route53_zones : !endswith(includedDomain, domain) if config.record_manager_role_arn == role])] // never exclude an ancestor of an included domain
-  } }
-
 }
 
-resource "random_id" "ids" {
```

Comment:
These ids are actually important to preserve as they indicate "ownership" over DNS records (see the TXT records in the DNS zone). If these get deleted, manual intervention of users will be necessary.

Ensure that `moved` blocks are included. See docs here: https://developer.hashicorp.com/terraform/language/modules/develop/refactoring#splitting-one-module-into-multiple.